java/jdbc/ConnectionSamples/ADBQuickStart.java

```diff
@@ -0,0 +1,149 @@
+      DB_PASSWORD = scanner.nextLine();
+    }
+    catch (Exception e) {
+       System.out.println("ADBQuickStart - Exception occurred : " + e.getMessage());
+    }
+    
+    // Get the PoolDataSource for UCP
+    PoolDataSource pds = PoolDataSourceFactory.getPoolDataSource();
+
+    // Set the connection factory first before all other properties
+    pds.setConnectionFactoryClassName(CONN_FACTORY_CLASS_NAME);
+    pds.setURL(DB_URL);
+    pds.setUser(DB_USER);
+    pds.setPassword(DB_PASSWORD);
+    pds.setConnectionPoolName("JDBC_UCP_POOL");
+
+    // Default is 0. Set the initial number of connections to be created
+    // when UCP is started.
+    pds.setInitialPoolSize(5);
+
+    // Default is 0. Set the minimum number of connections
+    // that is maintained by UCP at runtime.
+    pds.setMinPoolSize(5);
+
+    // Default is Integer.MAX_VALUE (2147483647). Set the maximum number of
+    // connections allowed on the connection pool.
+    pds.setMaxPoolSize(20);
+
+    // Get the database connection from UCP.
+    try (Connection conn = pds.getConnection()) {
+      System.out.println("Available connections after checkout: "
+          + pds.getAvailableConnectionsCount());
+      System.out.println("Borrowed connections after checkout: "
+          + pds.getBorrowedConnectionsCount());
+      // Perform a database operation
+      doSQLWork(conn);
+    }
+    
+    System.out.println("Available connections after checkin: "
+        + pds.getAvailableConnectionsCount());
+    System.out.println("Borrowed connections after checkin: "
+        + pds.getBorrowedConnectionsCount());
+  }  
+
+  /*
+   * Selects 20 rows from the SH (Sales History) Schema that is the accessible to all 
+   * the database users of autonomous database. 
+   */
+  public static void doSQLWork(Connection conn) {
+  
+    try {
+      conn.setAutoCommit(false);
+      // Prepare a statement to execute the SQL Queries.
+      try (Statement statement = conn.createStatement() ) {
+      // Select 20 rows from the CUSTOMERS table from SH schema. 
+      try (ResultSet resultSet = statement.executeQuery ("select CUST_ID,"  
+          +  "CUST_FIRST_NAME, CUST_LAST_NAME, CUST_CITY, CUST_CREDIT_LIMIT "
+          +  "FROM SH.CUSTOMERS WHERE ROWNUM < 20 order by CUST_ID ") ) {
+        System.out.println("\nCUST_ID" + " " + "CUST_FIRST_NAME" + " " + "CUST_LAST_NAME" 
+        + " " + "CUST_CITY" + " " + "CUST_CREDIT_LIMIT");
+        System.out.println("-----------------------------------------------------------");
+        while (resultSet.next()) {
+          System.out.println(resultSet.getString(1) + " " + resultSet.getString(2) + " " +
+           resultSet.getString(3)+ " " + resultSet.getString(4) + " " +
+           resultSet.getInt(5));
+        }
+        System.out.println("\nSuccessfully established a connection to Autonomous Database\n");
+       }
+      }
+    }
+    catch (SQLException e) {
```

Comment:
Make the method throw `SQLException` and promote the catch to the caller.

Reply:
Done